graphistry/plugins/sentinel.py

```diff
@@ -0,0 +1,699 @@
+                except Exception as exc:
+                    if unwrap_nested is True:
+                        raise RuntimeError(f"Failed to unwrap nested data: {exc}") from exc
+                    # Heuristic miss - fall back to flat table
+                    pass
+
+            # Default: flat table
+            if not result.column_names:
+                # Safety fallback
+                dfs.append(pd.DataFrame(result.data))
+            else:
+                dfs.append(pd.DataFrame(result.data, columns=result.column_names))
+
+        # Auto-unbox single table result if requested
+        if single_table:
+            if len(dfs) > 1:
+                logger.warning(f"Query returned {len(dfs)} tables, returning first table only")
+            return dfs[0]
+
+        return dfs
+
+    def kql_last(
+        self,
+        query: str,
+        *,
+        hours: float = 1,
+        **kwargs
+    ) -> Union[pd.DataFrame, List[pd.DataFrame]]:
+        """Execute KQL query for the last N hours.
+
+        Convenience wrapper for kql() that automatically sets the timespan
+        to the last N hours from now.
+
+        :param query: KQL query string to execute
+        :type query: str
+        :param hours: Number of hours to look back (default: 1)
+        :type hours: float
+        :param kwargs: Additional arguments passed to kql()
+        :returns: Query results as DataFrame(s)
+        :rtype: Union[pd.DataFrame, List[pd.DataFrame]]
+
+        **Example: Get security alerts from last 24 hours**
+            ::
+
+                import graphistry
+                g = graphistry.configure_sentinel(...)
+
+                alerts = g.kql_last('''
+                    SecurityAlert
+                    | project TimeGenerated, AlertName, Severity
+                    | order by TimeGenerated desc
+                ''', hours=24)
+
+        **Example: Get recent failed logins (last hour)**
+            ::
+
+                # Default is 1 hour
+                recent_failures = g.kql_last('''
+                    SecurityEvent
+                    | where EventID == 4625
+                    | summarize FailCount=count() by Account
+                ''')
+        """
+        return self.kql(query, timespan=timedelta(hours=hours), **kwargs)
+
+    def sentinel_tables(self) -> pd.DataFrame:
+        """List all available tables in the Log Analytics workspace.
+
+        :returns: DataFrame with table names
+        :rtype: pd.DataFrame
+
+        **Example**
+            ::
+
+                import graphistry
+                g = graphistry.configure_sentinel(...)
+
+                # Get list of all tables
+                tables = g.sentinel_tables()
+                print(f"Found {len(tables)} tables")
+                print(tables.head(10))
+        """
+        # Use Usage table to get all table names - this avoids union conflicts
+        query = """
+        Usage
+        | where TimeGenerated > ago(30d)
```

Comment:
`TimeGenerated` is surprising here?

If desired as a convenience, add an optional param with a sane default?